plugins/output/adapt/importsource.js

```diff
@@ -462,114 +452,132 @@ function ImportSource(req, done) {
+      var data = _.extend({}, originalData);
+      /**
+      * Basic prep of data
+      */
+      delete data._id;
+      delete data._trackingId;
+      delete data._latestTrackingId;
+      data.createdBy = app.usermanager.getCurrentUser()._id;
+      if(type !== 'course') {
+        data._courseId = courseId;
+      }
+      if(data._component) {
+        data._componentType = metadata.componentMap[data._component]._id;
+      }
+      if(data._parentId) {
+        if(metadata.idMap[data._parentId]) {
+          data._parentId = metadata.idMap[data._parentId];
+        } else {
+          detachedElementsMap[originalData._id] = type;
+          logger.log('warn', 'Cannot update ' + originalData._id + '._parentId, ' +  originalData._parentId + ' not found in idMap');
+          return resolve();
+        }
+      }
+      /**
+      * Content-specific attributes
+      */
+      if(type === 'course') {
+        try {
+          var contents = await fs.readFile(path.join(COURSE_ROOT_FOLDER, Constants.Folders.Source, Constants.Folders.Theme, plugindata.theme[0]._theme, 'less', Constants.Filenames.CustomStyle), 'utf8');
+          data = _.extend(data, { tags: formTags, customStyle: contents.toString() });
+        } catch(e) {
+          return reject(e);
+        }
+      }
+      else if(type === 'config') {
+        data = _.extend(data, enabledExtensions, plugindata.theme[0], plugindata.menu[0], { "_defaultLanguage" : COURSE_LANG });
+      }
+      /**
+      * Define the custom properties and and pluginLocations
+      */
+      var genericPropKeys = Object.keys(dbInstance.getModel(type).schema.paths);
+      var customProps = _.pick(data, _.difference(Object.keys(data), genericPropKeys));
 
-    // organise functions in series, mainly for readability
-    async.series([
-        function prepareData(cb) {
-          // basic prep of data
-          delete data._id;
-          delete data._trackingId;
-          delete data._latestTrackingId;
-          data.createdBy = app.usermanager.getCurrentUser()._id;
-          if(!_.isEmpty(courseId)) data._courseId = courseId;
-          if(data._component) {
-            data._componentType = metadata.componentMap[data._component]._id;
-          }
-          if(data._parentId) {
-            if(metadata.idMap[data._parentId]) {
-              data._parentId = metadata.idMap[data._parentId];
-            } else {
-              detachedElementsMap[originalData._id] = type;
-              logger.log('warn', 'Cannot update ' + originalData._id + '._parentId, ' +  originalData._parentId + ' not found in idMap');
-              return cb();
-            }
-          }
-          // define the custom properties and and pluginLocations
-          var genericPropKeys = Object.keys(dbInstance.getModel(type).schema.paths);
-          var customProps = _.pick(data, _.difference(Object.keys(data), genericPropKeys));
+      if(_.isEmpty(customProps)) resolve(data);
```

Comment:
Just checking you intended to remove the `return` here?

Reply:
Probably should still return to be safe.
```suggestion
      if(_.isEmpty(customProps)) return resolve(data);
```